backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/barchart/BpkBarChartModel.kt

```diff
@@ -0,0 +1,80 @@
+/*
+ * Backpack for Android - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.skyscanner.backpack.compose.barchart
+
+import androidx.annotation.FloatRange
+import androidx.compose.runtime.Immutable
+import androidx.compose.runtime.Stable
+
+/**
+ * Represents the view model used to provide data to the bar chart.
+ * @param caption a chart caption.
+ * @param items list of bar items to render.
+ * @param legend an optional legend.
+ */
+@Immutable
+data class BpkBarChartModel(
+  val caption: String,
+  val items: List<Item>,
+  val legend: Legend? = null,
+) {
+
+  /**
+   * Represents a single bar in the chart.
+   * @param key an unique identifier of the bar. This allows to save the selected element each time the model is updated.
+   * @param title a primary text placed just below the bar itself.
+   * @param subtitle a secondary text placed just below the title.
+   * @param badge text to be shown in the popup when the item is selected. If null, the item becomes inactive.
+   * @param group name of the group in which the item exists. The name is rendered above the bars and updated as the chart scrolls horizontally.
+   * @param value the value of the bar itself, should be a range between 0.0f and 1.0f. If it exceeds, it'll be clamped
+   */
+  @Immutable
+  data class Item(
+    @Stable val key: Any,
+    val title: String,
+    val subtitle: String,
+    val badge: String? = null,
+    val group: String,
+    @FloatRange(from = 0.0, to = 1.0) val value: Float,
```

Comment:
we treat the badge text missing as inactive, but an inactive item has no value either - could the two be a separate model that is nullable? otherwise it'll be confusing that you can set a value, but it has no effect

Reply:
It's expected from the design that no badge makes the columns to have a fixed height. However, I think we can do this – to have better API and less confusion.